colorbleed/scripts/fusion/switch_and_submit.py

```diff
@@ -144,22 +146,31 @@ def update_frame_range(comp, representations):
     fusion_lib.update_frame_range(start, end, comp=comp)
 
 
-def switch(asset_name, filepath=None, new=True):
+def switch(file_path=None, asset_name=None, new=True, deadline=True):
     """Switch the current containers of the file to the other asset (shot)
 
+    Exitcode 1: No item found to publish
+    Exitcode 2: Errors during publish process
+
     Args:
-        filepath (str): file path of the comp file
+        file_path (str): file path of the comp file
         asset_name (str): name of the asset (shot)
         new (bool): Save updated comp under a different name
+        deadline (bool): if set to true
 
     Returns:
         comp path (str): new filepath of the updated comp
 
     """
 
+    assert asset_name, "Function requires at least an asset name"
+
     # Ensure filename is absolute
-    if not os.path.abspath(filepath):
-        filepath = os.path.abspath(filepath)
+    if file_path:
+        file_path = os.path.abspath(file_path)
+
+    api.install(avalon.fusion)
```

Comment:
It feels odd to me that this function is installing the host. Shouldn't that be done outside of it - where we actually launch the host?

Hmm... that's just a thought. Seems to me that this function is just doing too much because I was looking at it just as a function one might call to switch whatever is in the current scene, but it's doing much more. It's opening comps, publishing jobs, etc. It's probably a remainder of it originally being the script itself.

Currently it's more a `load_comp_switch_asset_and_submit` kind of method which I'm not "expecting" from the function name and docstring.